nsxt/resource_nsxt_logical_router_uplink_port.go

```diff
@@ -0,0 +1,237 @@
+package nsxt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/vmware/go-vmware-nsxt/manager"
+	"log"
+	"net/http"
+)
+
+const LogicalRouterUpLinkPortResourceType = "LogicalRouterUpLinkPort"
+
+func resourceNsxtLogicalRouterUpLinkPort() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtLogicalRouterUpLinkPortCreate,
+		Read:   resourceNsxtLogicalRouterUpLinkPortRead,
+		Update: resourceNsxtLogicalRouterUpLinkPortUpdate,
+		Delete: resourceNsxtLogicalRouterUpLinkPortDelete,
+		Schema: map[string]*schema.Schema{
+			"revision": getRevisionSchema(),
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Description of this resource",
+				Optional:    true,
+			},
+			"display_name": {
+				Type:        schema.TypeString,
+				Description: "The display name of this resource. Defaults to ID if not set",
+				Optional:    true,
+				Computed:    true,
+			},
+			"tag": getTagsSchema(),
+			"logical_router_id": {
+				Type:        schema.TypeString,
+				Description: "Identifier for logical router on which this port is created",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"linked_logical_switch_port_id": {
+				Type:        schema.TypeString,
+				Description: "Identifier for the logical switch port to connect to",
+				Required:    true,
+			},
+			"edge_cluster_member_index": {
```

Comment:
for sake of consistency, please make the name plural, like in "edge_cluster_member_indexes" in nsxt_dhcp_server_profile. Perhaps schema can be reused (dhcp server profile imposes a limit of 2 indices, not sure the same limitation applies here)